Created Popup

wang-joseph opened the pull request:
Closes #10.

This popup will offer details about the photo, message, and title.

Todo:
- [x] made text area
- [x] made photo area
- [x] made it mobile-friendly
- [x] made photo area deal with landscape and portrait photos well
- [x] made exiting the popup
- [x] made clicking on a post open a popup

wang-joseph committed:
created popup text box

Popup text should now work properly and like the design in the figma. It should also be responsive.

"Mobile first" should indeed come first but I'll make adjustments for that.

wang-joseph committed:
created centered image that preserves aspect ratio

The image section to view the image is finished.

Mobile view should only contain this image.

wang-joseph committed:
Made the popup mobile-responsive

This was significantly less difficult than I imagined.

shari09 commented on src/components/home/Popup.tsx:
```diff
@@ -0,0 +1,114 @@
+import { jsx, SxStyleProp } from 'theme-ui';
+
+import { IPost } from '../../utils/interface';
+import { theme } from '../../utils/theme';
```
It's not needed to import `theme`

wang-joseph replied:
I removed that but I guess I didn't push it in this commit. Weird

shari09 commented on src/components/home/Popup.tsx:
```diff
@@ -0,0 +1,114 @@
+  }
+
+  const authorTextStyle: SxStyleProp = {
+    fontSize: 'fontSizes.medium',
+    fontFamily: 'fonts.body',
```
`fontSize: 'medium'` and `fontFamily: 'body'` is good, no need to have the `fonts.` in front of them

shari09 commented on src/components/home/Popup.tsx:
```diff
@@ -0,0 +1,114 @@
+      <div sx={photoRegionDiv}>
+        <div sx={borderStyle} />
+        <div sx={photoContainerStyle}> 
+          <img src={photo.url} alt="" sx={photoStyle} />
```
can you change `alt` to `photo.name`

shari09 commented on src/components/home/Popup.tsx:
```diff
@@ -0,0 +1,114 @@
+  return (
+    <div sx={popupDiv}>
+      <div sx={photoRegionDiv}>
+        <div sx={borderStyle} />
```
The photo region has a border? Or are you referring to the background of the photo region that's dark blue.

shari09 replied:
If so, rather than having two `div` before and after the image, just have the `photoRegionDiv` to be a flex box with that background colour and then center the image to the middle of the flex box.

wang-joseph replied:
That is a much better idea gimmie a second

wang-joseph replied:
Actually, rethinking about it, wouldn't having one flexbox for the entire photo area make `position: absolute` required to position the x button? Either that or you need to define another div for the flexbox above the photo container, which makes you make another div below the photo, which kinda ruins the point.

Also, is it even possible to declare certain parts of a flexbox transparent? If not, then there wouldn't be a transparent view of the images below. I'll keep it for now, and if you can correct me I'll gladly change it.

shari09 commented on src/components/home/Popup.tsx:
```diff
@@ -0,0 +1,114 @@
+        </div>
+        <div sx={borderStyle} />
+      </div>
+      <div sx={textRegionDiv}>
+        <h3 sx={authorTextStyle}>{name}</h3>
+        <div sx={lineStyle} />
+        <p>{message}</p>
+      </div>
```
Is this scrollable ~idk if it is~ 🤡

wang-joseph replied:
Nope! Not scrollable.

Scrollable text in that area feels a bit weird. I think it might be better if we simply shrunk the text size based on the screen size to prevent scrolling? I'll add scrolling as a backup though.

shari09 reviewed (changes requested):
Good luck~ 😄

wang-joseph committed:
Closing of the pop-up exists

wang-joseph committed:
Handles landscape and portrait photos

wang-joseph committed:
Added a scrollable text box for the message

Now the message can be scrolled in the event it overflows.

wang-joseph committed:
create popup from posts

Each post will make a popup appear that will cover the whole window and display details about the post. This popup appears on tap of the expand or photo, and not the entire wrapper as if it was only for wrapper and you were on mobile, you couldn't read more without opening the entire photo.

In mobile the popup only shows the photo.

wang-joseph committed:
Stopped main body scrolling once popup appears

wang-joseph committed:
Added fade-in animation for popups

EmeraldEntities commented:
This PR closes #10.

wang-joseph committed:
removed default export for Popup

shari09 commented on src/components/home/Popup.tsx:
```diff
@@ -0,0 +1,184 @@
+  // Prevent the body from scrolling in a popup
+  useEffect(() => {
+    // Haha, get it? Old overflow = oldverflow?
+    const oldverflow = document.body.style.overflow
```
haha

shari09 commented on src/pages/Home.tsx:
```diff
@@ -86,6 +83,34 @@ with me and we work things out well. our humours match and we're always on the s
 ur such an amazing person and an amazing boyfriend (ur mine >:)) that is just too rare. i love u and i'll always do so ;)`,
         photo: null,
       },
+      {
+        name: 'sift',
+        message: `hihi shari i guess this is sorta of a response message to your previous
+        passionate declaration of love but in case you haven't figured it out through
+        my various littered comments or this paragraph i too actually love you sosososo much.
+        I didn't actually realize how much i loved you and i sitll cant believe its possible to
+        have feelings of affection this strong towards someone heheh.
+        its just so much fun whispering i love you and seeing you flinch or staring at your cute being
+        absentmindedly when im killing time. look at you always makes me smile so much and im really glad
+        you're so happy as a result :))
+        i really like how you're so go-getting and responsible and serious/charismatic and all leaderlike at times
+        but you always have this sweet and sappy and adorable side to you that i absolutely adore now
+        (although i wasn't good in the past but even then you still stuck with me and made me a better person ahhhh).
+        we work out so well together and we can actually communicate even if we
+        fight and stuff, and we always become sooo lovey after anything. you know be a bit tooo well its almost really
+        scary :p. your compliments make me feel warm and fuzzy inside and want to cherish you moreee.
+        you're such an amazing person and an amazing girlfriend and here's to us sticking together forever! i love you, shari.
+        i really really really love you :)
+        
+        ps... was that a confession???
+        pps... photo unrelated i just wanted to test landscape hehehehe
+        ppps... i hope this message is so long that i get to test scroll
+        pppps... i love u!!!`,
```
awwwwwwwwwwwwwwwwwwwww ❤️

shari09 replied:
love u ❤️ 😊

shari09 commented on src/components/home/Popup.tsx:
```diff
@@ -0,0 +1,184 @@
+
+  return (
+    <div sx={popupDiv}>
+      <div sx={photoRegionDiv} onClick={closeHandler}>
+        <div sx={borderStyle}>
+          <RiCloseCircleLine color='white' size={30} sx={returnButtonStyle} onClick={closeHandler}/>
+        </div>
+        <div sx={photoContainerStyle}> 
+          <img src={photo.url} alt={photo.name} sx={photoStyle} onClick={consumeMouseInteraction} />
+        </div>
+        <div sx={borderStyle} />
+      </div>
```
Why don't you make the containing div (`photoRegionDiv`) background dark blue and have it close if you click anywhere that's not the photo itself. Right now, it doesn't close a landscape photo unless you close on the border (nothing happens when you click on the extra space).

wang-joseph replied:
Writing an issue for this.

shari09 commented on src/components/home/Popup.tsx:
```diff
@@ -0,0 +1,184 @@
+    overflowY: 'auto',
+    width: '100%',    
+    maxHeight: '85%',
+    '&::-webkit-scrollbar': {
+      width: '0.5em'
+    },
+    '&::-webkit-scrollbar-thumb': {
+      backgroundColor: 'rgba(0, 0, 0, 0.05)',
+      outline: '1px solid slategrey'
+    }
```
Can you make the scrollbar to not have outlines and perhaps rounded corners

wang-joseph replied:
It's a temp scroll bar, I wanted to get this push out now and not focus too much on the very little details (issue time?)

shari09 commented on src/components/home/Popup.tsx:
```diff
@@ -0,0 +1,184 @@
+    evt.stopPropagation();
+  }
+
+  // In the event that this popup is created somehow with no photo, generate
+  // empty element instead of throwing an error.
+  if (!photo) {
+    return <React.Fragment></React.Fragment>
+  }
```
rip message only popups that have a looooooooooooong message😢 

lol I'll create a new issue for this since it's gonna be quite a change

shari09 replied:
nevermind I realized that all you do is just conditionally display the photo

shari09 commented on src/components/home/Post.tsx:
```diff
@@ -2,8 +2,11 @@
 import React, { useEffect, useRef, useState } from 'react';
 import {jsx, SxStyleProp} from 'theme-ui';
 import {AiOutlineExpandAlt} from 'react-icons/ai';
```
Just a style thing, are we doing space or no space for curly brackets

wang-joseph replied:
We can refactor the styles later but I guess we'll do space?

shari09 commented on src/components/home/Post.tsx:
```diff
@@ -162,20 +163,37 @@ export const Post: React.FC<PostProps> = React.memo(({post, style}) => {
   }, [nameRef.current, messageRef.current]);
 
-  
+  /**
+   * Displays the details of this post in a popup.
+   */
+  const popup = (): void => {
+    setShowPopup(true);
+  };
+
+  /**
+   * Closes the shown popup, if any.
+   */
+   const closePopup = (): void => {
+    setShowPopup(false);
+  }
+
```
Why don't you just use anonymous functions when needing them since all it does it set a state true/false

no need to scroll up to see what it does when these functions are called

wang-joseph replied:
I think these functions already originally had longer values but now they don't anymore and I just forgot to turn them into anonymous functions

shari09 commented on src/pages/Home.tsx:
```diff
@@ -179,9 +199,8 @@ ur such an amazing person and an amazing boyfriend (ur mine >:)) that is just to
     );
   };
 
-
   return (
-    <div sx={wrapperStyle}>
+    <div sx={wrapperStyle}> 
```
nice change

wang-joseph committed:
converted oneliner functions to anonymous functions
